### PR TITLE
Fix bottomSheet bottom insets

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR fixes the all the bottom sheets bottom insets, particularly visible on iOS devices.

Now we don't need to pass the inset to the function or styles but only to the library BottomSheet component as a prop, which also simplifies a lot of code.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55073

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]

File: app/products/calls/hooks.ts [hunk 1]
```diff
@@ -217,7 +217,7 @@ export const useHostMenus = () => {
         const props = {closeButtonId: closeUserProfile, location: '', userId: session.userId};
 
         openAsBottomSheet({screen, title, theme, closeButtonId: closeUserProfile, props});
-    }, [theme, currentCall?.channelId]);
+    }, [intl, theme]);
```
Comment (cpoile) [inline app/products/calls/hooks.ts:220]: thanks!

**Review by cpoile — APPROVED** [review]
Nice, and thanks for the fixes

**Comment by lindy65** [comment]
Hi @enahum - please let me know if the screenshots are correct and what is expected

Bottom sheet expanded
image

Bottom sheet normal
image

**Comment by enahum** [comment]
@lindy65  yup, that is fine.. you can also try with up to 3 teams and see the difference.

**Comment by enahum** [comment]
@lindy65 btw even if the ticket does not mention it.. this PR is affecting ALL the bottom sheets across the app.

**Comment by lindy65** [comment]
Hey @enahum 

Testing on Android, the first screenshot is first time open from Search and tapping on the Team icon (it doesn't change if I close and reopen)
image

The second screenshot is if I try to swipe up on the bottom sheet to maximise it - it doesn't change size but rather scrolls the contents of the bottom sheet up
image

Does this look correct to you? Behavior in the bottom sheet is different to iOS/iPhone as there, the bottom sheet is maximisable when swiping up on it whereas Android the size of the bottom sheet doesn't change but the contents scroll

**Comment by enahum** [comment]
@lindy65 on Android that is not correct.. as I mentioned before please check **ALL** bottom sheets across the entire app!!!!!!!

That said, in the meantime, I'll check if I get the same results on Android. can you share what version of android are you on?

**Comment by lindy65** [comment]
Yep, I am testing the PR on all mobile platforms @enahum 

I'm on Samsung S22, Android v14

**Comment by lindy65** [comment]
Ooooh, you mean ALL bottom sheets - not just THIS bottom sheet across platforms - got it now :)

**Comment by enahum** [comment]
> Yep, I am testing the PR on all mobile platforms @enahum
> 
> I'm on Samsung S22, Android v14

ok that is good,.. but is ALL bottom sheets.. reactions, participants, the plus sign in the home screen, multi-server, three dots in the channel screen, announcement banner, add bookmarks, file search results, long press on at mentions, long press on links, post options, long press on markdown images, pressing on the camera icon when composing a message, post priority, retry a failed post, audio devices for calls, host controls for calls, browse channels (when you tap on switching the channel type), etc.. 

if you look at the code, the name of the files should give you an idea

**Comment by enahum** [comment]
> Hey @enahum
> 
> Testing on Android, the first screenshot is first time open from Search and tapping on the Team icon (it doesn't change if I close and reopen) image
> 
> The second screenshot is if I try to swipe up on the bottom sheet to maximise it - it doesn't change size but rather scrolls the contents of the bottom sheet up image
> 
> Does this look correct to you? Behavior in the bottom sheet is different to iOS/iPhone as there, the bottom sheet is maximisable when swiping up on it whereas Android the size of the bottom sheet doesn't change but the contents scroll

@lindy65  this one should be fixed with the latest commit.

**Comment by lindy65** [comment]
> > Hey @enahum
> > Testing on Android, the first screenshot is first time open from Search and tapping on the Team icon (it doesn't change if I close and reopen) image
> > The second screenshot is if I try to swipe up on the bottom sheet to maximise it - it doesn't change size but rather scrolls the contents of the bottom sheet up image
> > Does this look correct to you? Behavior in the bottom sheet is different to iOS/iPhone as there, the bottom sheet is maximisable when swiping up on it whereas Android the size of the bottom sheet doesn't change but the contents scroll
> 
> @lindy65 this one should be fixed with the latest commit.

Thanks @enahum - this is indeed fixed with the latest commit.

**Review by lindy65 — APPROVED** [review]
Thanks @enahum - I have tested all the bottom sheets according to files changed and they look good to me - I didn't find any issues apart from the issue on Android which you have already fixed.

**Comment by enahum** [comment]
@matthewbirtch can you provide measurements instead of some space ?

why do I ask, cause everything is the same, just move the `bottom` inset to another more centralized placed in the code.

**Comment by matthewbirtch** [comment]
👇 add 12 space please on Android
image

👇 For the message priority bottom sheet, it looks good on iPhone/Android phone now, but the iPad now has too much space in the button footer. It should be reduced for iPad by 12.
image

👇 For the DM selection bottom container, remove 20 below the button
image


With the latest update, I just noticed that you fixed the Android search bottom sheet which is great, but on iOS now a scrollbar shows if there are 3 teams. I know anything beyond 3 teams can be visible by expanding the bottom sheet, but I think the scroll bar for 3 teams isn't intended.

https://github.com/user-attachments/assets/789b1a66-95bc-4d8f-8b7f-674f7458425a

**Comment by enahum** [comment]
@matthewbirtch changes made :)

**Review by matthewbirtch — APPROVED** [review]
Looks great @enahum. I'm going to approve. Only one minor update on iPad:
- The space at the bottom of the message priority modal is now too small (was too big previously). It should a space of 20, but looks to be 12 on iPad (only on iPad)
- Screenshot 2024-11-28 at 12 12 34 PM

**Pull request merged**